File: ayrfu-frontend/src/api/services/serviceRequest.service.js
import axios from '../axios';

const serviceRequestService = {
  createServiceRequest: async (requestData) => {
    return axios.post('/service-requests', requestData);
  },

  getServiceRequestById: async (id) => {
    return axios.get(`/service-requests/${id}`);
  },

  getServiceRequestsByClient: async (clientId, page = 0, size = 20) => {
    return axios.get(`/service-requests/clients/${clientId}?page=${page}&size=${size}`);
  },

  getCurrentClientServiceRequests: async (page = 0, size = 20) => {
    return axios.get(`/service-requests/my-requests?page=${page}&size=${size}`);
  },

  getServiceRequestsByStatus: async (status, page = 0, size = 20) => {
    return axios.get(`/service-requests/status/${status}?page=${page}&size=${size}`);
  },

  updateServiceRequestStatus: async (id, status) => {
    return axios.patch(`/service-requests/${id}/status?status=${status}`);
  },

  deleteServiceRequest: async (id) => {
    return axios.delete(`/service-requests/${id}`);
  }
};

export default serviceRequestService;

File: ayrfu-backend/src/main/java/com/ayrfu/ayrfu/service/ServiceRequestService.java
package com.ayrfu.ayrfu.service;

import com.ayrfu.ayrfu.dto.request.ServiceRequestRequest;
import com.ayrfu.ayrfu.dto.response.ServiceRequestResponse;
import com.ayrfu.ayrfu.enumeration.ServiceRequestStatus;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

public interface ServiceRequestService {


    @Transactional
    ServiceRequestResponse createServiceRequest(ServiceRequestRequest requestDto);

    @Transactional(readOnly = true)
    ServiceRequestResponse getServiceRequestById(Long id);

    @Transactional(readOnly = true)
    List<ServiceRequestResponse> getServiceRequestsByClient(Long clientId, int page, int size);

    @Transactional(readOnly = true)
    List<ServiceRequestResponse> getCurrentClientServiceRequests(int page, int size);

    @Transactional(readOnly = true)
    List<ServiceRequestResponse> getServiceRequestsByStatus(ServiceRequestStatus status, int page, int size);

    @Transactional
    ServiceRequestResponse updateServiceRequestStatus(Long id, ServiceRequestStatus status);

    @Transactional
    void deleteServiceRequest(Long id);

    boolean isOwnRequest(Long requestId, Long userId);
}